compaction/compaction.cc

```diff
@@ -75,6 +75,16 @@ static const std::unordered_map<compaction_type, sstring> compaction_types = {
     { compaction_type::Reshape, "RESHAPE" },
 };
 
+template <>
+std::string cm_format<with_origin::yes>(const shared_sstable& sst) {
+    return fmt::format("{:D}:level={:d}:origin={}", *sst, sst->get_sstable_level(), sst->get_origin());
```

Comment:
how about adding yet another specifier to the formatter? like
```c++
sstables::component_type component_type = sstables::component_type::Data;
bool with_level = false;
bool with_origin = false;

constexpr auto parse(format_parse_context& ctx) {
  auto it = ctx.begin(), end = ctx.end();
  if (it != end) {
    switch (*it) {
      case 'T':
        component_type = sstables::component_type::Data;
        ++it;
        break;
      // ..
    }
  }
  if (it != end) {
    switch (*it) {
      case 'L':
        with_level = true;
        with_origin = true;
        ++it;
        break;
      // ..
    }
  }
  if (it != end && *it != '}') {
    throw format_error("invalid format");
  }
}
```

```c++
fmt::format("{:DL}", sstable);

Reply:
Right now I see no point in it as it's only compaction manager than needs this and can go with explicit `sstring cm_format(sstable)` helper